### PR TITLE
Sample Datasets

### PR DESCRIPTION
**Pull request by Davev4d3**
**Proposed changes**:
- Sample datasets can now be found in the folder `sample_datasets`, they're from the [Snips Benchmark Repository](https://github.com/snipsco/nlu-benchmark)
- A simple script to convert from Snips format to Rasa's format.

**Status**:
- [x] Sample datasets in Rasa's format
- [x] Conversion script

**P.S.** I'm a student and I'm still learning, so the python code may not be _top quality_. Please feel free to give feedback and make improvements or just drop by and say hi =)

### CODE REVIEW

File: sample_datasets/snips-to-rasa.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import json
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: please import the standart future imports:

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
```

File: sample_datasets/snips-to-rasa.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import json
+
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: Please rename to `snips_to_rasa.py`

File: sample_datasets/snips-to-rasa.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import json
+
+inputFiles = [
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: please wrap code in `if __name__ == '__main__':`

File: sample_datasets/snips-to-rasa.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    '2017-06-custom-intent-engines/BookRestaurant/train_BookRestaurant_full.json',
+    '2017-06-custom-intent-engines/GetWeather/train_GetWeather_full.json',
+    '2017-06-custom-intent-engines/PlayMusic/train_PlayMusic_full.json',
+    '2017-06-custom-intent-engines/RateBook/train_RateBook_full.json',
+    '2017-06-custom-intent-engines/SearchCreativeWork/train_SearchCreativeWork_full.json',
+    '2017-06-custom-intent-engines/SearchScreeningEvent/train_SearchScreeningEvent_full.json',
+]
+
+for inputFile in inputFiles:
+    print('Now converting: ' + inputFile)
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: please use a logge, so at the beginning of the file 

```
import logging

logger = logging.getLogger(__name__)
```

and then `logger.debug(...)`

File: sample_datasets/snips-to-rasa.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    with open(inputFile, 'r', encoding='utf8') as f:
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: `io.open` instead of `open`

File: sample_datasets/snips-to-rasa.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+            text = ''
+            textLength = 0
+            entities = []
+            exampleLength = len(example) - 1
+
+            for index, phrase in enumerate(example):
+                # print(index, exampleLength)
+
+                if index != exampleLength:  # or (index == exampleLength and phrase['text'] != ' '):
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: whats the comment?

File: sample_datasets/snips-to-rasa.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                "intent": intent,
+                "entities": entities
+            })
+
+        outputRASA = {
+            "rasa_nlu_data": {
+                "common_examples": outputExamples
+            }
+        }
+        outputFile = open(intent + '.json', 'w', encoding='utf8')
```
Comment (tmbo) [inline sample_datasets/snips-to-rasa.py]: please wrap in `with`, e.g.

```
with io.open(intent + ".json", 'w', encoding='utf8') as f:
  f.write(json.dumps(outputRASA, sort_keys=False, indent=2))

File: sample_datasets/snips-to-rasa.py (hunk 6 above)
Comment (Davev4d3) [reply sample_datasets/snips-to-rasa.py]: Some datasets had strange spaces at the end of some sentences, so I made a _really_ dodgy workaround

**Comment by Davev4d3** [comment]
Hey @tmbo 
That's okay. As for the conversion script, I don't think there is much point for users to run/use it. I included it mainly for reference. The datasets are already here in rasa format, so we can safely dispose of the conversion script.

**Comment by tmbo** [comment]
Sounds good. Here is what would be good: can you post the script in a comment on this pr attached as a file? After that we can safely remove it from the PR and still have it if we happen to need it in the future.

**Comment by Davev4d3** [comment]
`.py` files aren't supported as attachment, so here's a zip. The file _hasn't_ been updated, as there is no need for the conversion script yet.
[snips_to_rasa.zip](https://github.com/RasaHQ/rasa_nlu/files/1562095/snips_to_rasa.zip)

**Comment by tmbo** [comment]
I really appreciate the effort and time you put into this. Nevertheless, after talking to the other contributors I don't think this should go into the main repository. To keep the footprint small I'd either like to put it into a separate repository or link it from the documentation.

**Comment by Davev4d3** [comment]
Yeah mate thats fine. How would you suggest we proceed?

**Pull request closed without merging**